UpdatedImageProcessing/TargetDetection/blob_color_operations.py

```diff
@@ -0,0 +1,143 @@
+
+        left_color_1 = target_map_image.load()[blob_location[0] + (blob_location[2] / 6), blob_location[1] + (blob_location[3] / 2)]
+        left_color_2 = target_map_image.load()[blob_location[0] + (blob_location[2] / 3), blob_location[1] + (blob_location[3] / 2)]
+
+        right_color_1 = target_map_image.load()[blob_location[0] + (blob_location[2] / 6 * 5), blob_location[1] + (blob_location[3] / 2)]
+        right_color_2 = target_map_image.load()[blob_location[0] + (blob_location[2] / 3 * 2), blob_location[1] + (blob_location[3] / 2)]
+
+        left_upper_color_1 = target_map_image.load()[blob_location[0] + (blob_location[2] / 6), blob_location[1] + (blob_location[3] / 6)]
+        left_upper_color_2 = target_map_image.load()[blob_location[0] + (blob_location[2] / 3), blob_location[1] + (blob_location[3] / 6)]
+        left_upper_color_3 = target_map_image.load()[blob_location[0] + (blob_location[2] / 6), blob_location[1] + (blob_location[3] / 3)]
+        left_upper_color_4 = target_map_image.load()[blob_location[0] + (blob_location[2] / 3), blob_location[1] + (blob_location[3] / 3)]
+
+        right_upper_color_1 = target_map_image.load()[blob_location[0] + (blob_location[2] / 3 * 2), blob_location[1] + (blob_location[3] / 6)]
+        right_upper_color_2 = target_map_image.load()[blob_location[0] + (blob_location[2] / 6 * 5), blob_location[1] + (blob_location[3] / 6)]
+        right_upper_color_3 = target_map_image.load()[blob_location[0] + (blob_location[2] / 3 * 2), blob_location[1] + (blob_location[3] / 3)]
+        right_upper_color_4 = target_map_image.load()[blob_location[0] + (blob_location[2] / 6 * 5), blob_location[1] + (blob_location[3] / 3)]
+
+        left_lower_color_1 = target_map_image.load()[blob_location[0] + (blob_location[2] / 6), blob_location[1] + (blob_location[3] / 3 * 2)]
+        left_lower_color_2 = target_map_image.load()[blob_location[0] + (blob_location[2] / 3), blob_location[1] + (blob_location[3] / 3 * 2)]
+        left_lower_color_3 = target_map_image.load()[blob_location[0] + (blob_location[2] / 6), blob_location[1] + (blob_location[3] / 6 * 5)]
+        left_lower_color_4 = target_map_image.load()[blob_location[0] + (blob_location[2] / 3), blob_location[1] + (blob_location[3] / 6 * 5)]
+
+        right_lower_color_1 = target_map_image.load()[blob_location[0] + (blob_location[2] / 3 * 2), blob_location[1] + (blob_location[3] / 3 * 2)]
+        right_lower_color_2 = target_map_image.load()[blob_location[0] + (blob_location[2] / 6 * 5), blob_location[1] + (blob_location[3] / 3 * 2)]
+        right_lower_color_3 = target_map_image.load()[blob_location[0] + (blob_location[2] / 3 * 2), blob_location[1] + (blob_location[3] / 6 * 5)]
+        right_lower_color_4 = target_map_image.load()[blob_location[0] + (blob_location[2] / 6 * 5), blob_location[1] + (blob_location[3] / 6 * 5)]
```

Comment:
Because you use the load function so often, it would make sense to save some computation time by creating a variable called something like ```pixel_access_target_map_image``` which is the loaded ```target_map_image```. This way, you only have to load the image one time, instead of the 25 times used currently.